### PR TITLE
Adding aws_batch_job_definition to v1beta1 version

### LINKED ISSUE #162: batch: aws_batch_job_definition CannotObserveExternalResource deployment error
### What happened?

While testing resource `aws_batch_job_definition` we got an error in `make run` command.

```
Message: observe failed: cannot run refresh: refresh failed: "name" ("test:1") must be up to 128 letters (uppercase and lowercase), numbers, underscores and dashes, and must start with an alphanumeric.: 
```

This issue relates to https://github.com/upbound/provider-aws/issues/200

Related files:
[jobdefinition.log](https://github.com/upbound/provider-aws/files/10042527/jobdefinition.log)
[jobdefinition.txt](https://github.com/upbound/provider-aws/files/10042528/jobdefinition.txt)

### How can we reproduce it?

Copy generated manifest and try to apply.
### What environment did it happen in?

* Universal Crossplane Version:
* Provider Version: AWS main branch head

### PR DESCRIPTION
**Pull request by huffmanjohnf**
### Description of your changes
Adding (1) resources in the batch group:

 - [aws_batch_job_definition](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/batch_job_definition)

Fixes #162
Related to #200 

I have:

- [X] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Applied in personal AWS account using local deployment created by `make local-deploy`.

### CODE REVIEW

**Comment by huffmanjohnf** [comment]
/test-examples="examples/batch/jobdefinition.yaml"

**Comment by jeanduplessis** [comment]
/test-examples="examples/batch/jobdefinition.yaml"

File: config/externalname.go [hunk 1]
```diff
@@ -1546,6 +1546,9 @@ var ExternalNameConfigs = map[string]config.ExternalName{
 	//
 	// Batch Scheduling Policy can be imported using the arn: arn:aws:batch:us-east-1:123456789012:scheduling-policy/sample
 	"aws_batch_scheduling_policy": config.TemplatedStringAsIdentifier("name", "arn:aws:batch:{{ .setup.configuration.region }}:{{ .setup.client_metadata.account_id }}:scheduling-policy/{{ .external_name }}"),
+	// Batch Job Definition can be imported using ARN that has a random substring, revision at the end:
+	// arn:aws:batch:us-east-1:123456789012:job-definition/sample:1
+	"aws_batch_job_definition": config.IdentifierFromProvider,
```
Comment (turkenf) [inline config/externalname.go:1551]:
```suggestion
	"aws_batch_job_definition": config.TemplatedStringAsIdentifier("name", "arn:aws:batch:{{ .setup.configuration.region }}:{{ .setup.client_metadata.account_id }}:job-definition/{{ .external_name }}"),
```
If you haven't tried it, can you please try the configuration above?

**Review by turkenf — COMMENTED** [review]
Thank you for this PR @huffmanjohnf, I left a comment to consider.

File: config/externalname.go (hunk 1 above)
Comment (huffmanjohnf) [reply config/externalname.go:1551]: Thank you for the suggestion @turkenf. That was the first configuration I attempted, which ended up reproducing the behavior described in #162.

In summary, the configuration you suggested would successfully create the external resource but then the controller could not observe it afterwards due to:
> Message: observe failed: cannot run refresh: refresh failed: "name" ("test:1") must be up to 128 letters (uppercase and lowercase), numbers, underscores and dashes, and must start with an alphanumeric.:  

In other words, the external name part of the ARN for a job-definition has a name (defined in config, i.e. `test`) and a version (provided by AWS, i.e. `:1`).

If I understood the [Contributing docs](https://github.com/upbound/upjet/blob/main/docs/add-new-resource-short.md) correctly, the `version` suffix provided by AWS breaks the pattern for [Case 5: Non-random Substrings as Identifier](https://github.com/upbound/upjet/blob/main/docs/add-new-resource-short.md#case-5-non-random-substrings-as-identifier) used for most ARNs and instead falls into the random substring case:
> However, there are cases where the ARN includes random substring and that would fall under [Case 4](https://github.com/upbound/upjet/blob/main/docs/add-new-resource-short.md#case-4-random-identifier-substring-from-provider).

I ended up choosing `config.IdentifierFromProvider` based on other resources with ARNs following the `name:version` pattern, such as an [`aws_ecs_task_definition` external name](https://github.com/upbound/provider-aws/blob/main/config/externalname.go#L379-L381). This worked locally, as I could create and observe the external resource from a local Crossplane cluster (created by `make local-deploy`) against my own AWS account.

I've never worked with golang before and mostly relied on the docs and Makefile to create this PR, so I'm certainly open for guidance on the best way to configure this. I am not sure how to troubleshoot the automated test logs or the error (`uptest-v0.5.0: error: cannot run e2e tests successfully: cannot execute tests: kuttl failed: exit status 1`) and can't seem to get `make e2e` run on my machine. Any help is much appreciated!

**Comment by turkenf** [comment]
/test-examples="examples/batch/jobdefinition.yaml"

**Comment by turkenf** [comment]
@huffmanjohnf, uptest failed because of:
```
ClientException: Error executing request, Exception : Value 0.25 for type VCPU in resourceRequirement is not valid. Please provide a numeric value.
```

**Comment by huffmanjohnf** [comment]
Awesome, thank you for sharing the exception! That points me in the right direction. I'll update the `resourceRequirement` value from the Terraform example.

Where did you find that error trace? I didn't see it in the action logs.

**Comment by turkenf** [comment]
Please see: https://github.com/upbound/provider-gcp/pull/391#issuecomment-1725071475

**Comment by huffmanjohnf** [comment]
Ahh, I see. Thank you! 
Ok, I've updated `examples/batch/jobdefinition.yaml` with a vCPU value of `"1"`. I guess the [example in the Terraform docs](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/batch_job_definition#example-usage) is no longer valid as vCPU must be an integer greater than or equal to 1. I left the original example `examples-generated/batch/jobdefinition.yaml` untouched.

**Comment by huffmanjohnf** [comment]
/test-examples="examples/batch/jobdefinition.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/batch/jobdefinition.yaml"

**Review by turkenf — APPROVED** [review]
Thank you for your contribution @huffmanjohnf, LGTM.

**Pull request merged**